File: codes/java/chap06/answer_A39.java
import java.util.*;

class Main {
	public static void main(String[] args) {
		Scanner sc = new Scanner(System.in);

		// 입력
		int N = sc.nextInt();
		int[][] A = new int[N][2];
		for (int i = 1; i <= N; i++) {
			A[i - 1][0] = sc.nextInt();
			A[i - 1][1] = sc.nextInt();
		}

		// 배열 A를 종료 시각(A[x][1]) 기준 오름차순 정렬
		Arrays.sort(A, (a1, a2) -> Integer.compare(a1[1], a2[1]));

		// 종료 시각이 빠른 것부터 탐욕적으로 취해 간다(CurrentTime은 현재 시각)
		int CurrentTime = 0;
		int Answer = 0;
		for (int i = 1; i <= N; i++) {
			if (CurrentTime <= A[i - 1][0]) {
				CurrentTime = A[i - 1][1];
				Answer += 1;
			}
		}

		// 출력
		System.out.println(Answer);
	}
};

File: codes/python/chap06/answer_A39.py
# 입력
# A는 (종료 시작, 시작 시각)으로 되어 있는 것에 주의[종료 시각 기준 오름차순 정렬을 위해]
N = int(input())
A = []
for i in range(N):
	l, r = map(int, input().split())
	A.append([r, l])

# 정렬
A.sort()

# 종료 시각이 빠른 것부터 탐욕적으로 취해 간다(CurrentTime은 현재 시각)
CurrentTime = 0
Answer = 0
for i in range(N):
	if CurrentTime <= A[i][1]:
		CurrentTime = A[i][0]
		Answer += 1

# 출력
print(Answer)
